engine/runtime/src/main/java/org/enso/interpreter/runtime/EnsoContext.java

```diff
@@ -399,18 +423,63 @@ public Object lookupJavaClass(String className) {
       List<String> nestedClassPart =
           i < items.size() - 1 ? items.subList(i + 1, items.size()) : List.of();
       try {
-        Object hostSymbol = environment.lookupHostSymbol(pkgName + "." + curClassName);
+        var hostSymbol = lookupHostSymbol(pkgName, curClassName);
         if (nestedClassPart.isEmpty()) {
           return hostSymbol;
         } else {
-          return getNestedClass(hostSymbol, nestedClassPart);
+          var fullInnerClassName = curClassName + "$" + String.join("$", nestedClassPart);
+          return lookupHostSymbol(pkgName, fullInnerClassName);
         }
-      } catch (RuntimeException ignored) {
+      } catch (RuntimeException | InteropException ex) {
+        logger.log(Level.WARNING, null, ex);
       }
     }
     return null;
   }
 
+  private Object lookupHostSymbol(String pkgName, String curClassName)
+  throws UnknownIdentifierException, UnsupportedMessageException {
+    if (findGuestJava() == null) {
+      return environment.lookupHostSymbol(pkgName + "." + curClassName);
+    } else {
+      return InteropLibrary.getUncached().readMember(findGuestJava(), pkgName + "." + curClassName);
+    }
+  }
+
+  private Object guestJava = this;
+
+  @TruffleBoundary
+  private Object findGuestJava() throws IllegalStateException {
+    if (guestJava != this) {
+      return guestJava;
+    }
+    guestJava = null;
+    var envJava = System.getenv("ENSO_JAVA");
+    if (envJava == null) {
+      return guestJava;
+    }
+    if ("espresso".equals(envJava)) {
+      var src = Source.newBuilder("java", "<Bindings>", "getbindings.java").build();
+      try {
+        guestJava = environment.parsePublic(src).call();
+        logger.log(Level.SEVERE, "Using experimental Espresso support!");
+      } catch (Exception ex) {
+        if (ex.getMessage().contains("No language for id java found.")) {
+          logger.log(Level.SEVERE, "Environment variable ENSO_JAVA=" + envJava + ", but " + ex.getMessage());
+          logger.log(Level.SEVERE, "Use " + System.getProperty("java.home") + "/bin/gu install espresso");
```

Comment:
Sometimes we check for `JAVA_HOME` and sometimes for `java.home`. Sometimes for both.

Reply:
We should be using `System.getProperty("java.home")` as that it guaranteed to be set on OpenJDK. `JAVA_HOME` is just a convenience environment variable used by some (usually shell) scripts to locate the `java` command.